File: Backend/HRMS/HRMS.Application/DTOs/Core/JobGradeDto.cs
namespace HRMS.Application.DTOs.Core;

public class JobGradeDto
{
    public int JobGradeId { get; set; }
    public string GradeCode { get; set; } = string.Empty;
    public string GradeNameAr { get; set; } = string.Empty;
    public string GradeNameEn { get; set; } = string.Empty;
    public int GradeLevel { get; set; }
    public decimal MinSalary { get; set; }
    public decimal MaxSalary { get; set; }
    public string? BenefitsConfig { get; set; }
    public string? Description { get; set; }
}

public class JobGradeListDto
{
    public int JobGradeId { get; set; }
    public string GradeCode { get; set; } = string.Empty;
    public string GradeNameAr { get; set; } = string.Empty;
    public string GradeNameEn { get; set; } = string.Empty;
    public int GradeLevel { get; set; }
    public decimal MinSalary { get; set; }
    public decimal MaxSalary { get; set; }
}

File: Frontend/HRMS_Frontend/src/app/features/setup/models/job-grade.dto.ts
// Matches HRMS.Application.DTOs.Core.JobGradeDto
export interface JobGradeDto {
  jobGradeId: number;
  gradeCode: string;
  gradeNameAr: string;
  gradeNameEn: string;
  gradeLevel: number;
  minSalary: number;
  maxSalary: number;
  benefitsConfig?: string | null;
  description?: string | null;
}

// Matches HRMS.Application.Features.Core.JobGrades.Commands.CreateJobGrade.CreateJobGradeCommand
export interface CreateJobGradeCommand {
  gradeCode: string;
  gradeNameAr: string;
  gradeNameEn: string;
  gradeLevel: number;
  minSalary: number;
  maxSalary: number;
  benefitsConfig?: string | null;
  description?: string | null;
}

// Matches HRMS.Application.Features.Core.JobGrades.Commands.UpdateJobGrade.UpdateJobGradeCommand
export interface UpdateJobGradeCommand {
  jobGradeId: number;
  gradeCode: string;
  gradeNameAr: string;
  gradeNameEn: string;
  gradeLevel: number;
  minSalary: number;
  maxSalary: number;
  benefitsConfig?: string | null;
  description?: string | null;
}
